src/cbc_sdk/platform/alerts.py

```diff
@@ -1510,3 +1632,184 @@ def set_remote_is_private(self, is_private, exclude=False):
+        """
+        request = super(GroupedAlertSearchQuery, self)._build_request(from_row, max_rows, add_sort=True)
+        request["group_by"] = {"field": self._group_by}
+
+        return request
+
+    def get_alert_search_query(self):
+        """
+        Converts the GroupedAlertSearchQuery into a nongrouped AlertSearchQuery
+
+        Returns: AlertSearchQuery
+
+        Note: Does not preserve sort criterion
+        """
+        alert_search_query = self._cb.select(Alert)
+        for key, value in vars(alert_search_query).items():
+            if hasattr(self, key) and key not in REQUEST_IGNORED_KEYS:
+                setattr(alert_search_query, key, self.__getattribute__(key))
+
+        return alert_search_query
+
+    def _perform_query(self, from_row=1, max_rows=-1):
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): The row to start the query at (default 1).
+            max_rows (int): The maximum number of rows to be returned (default -1, meaning "all").
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url("/_search")
+        current = from_row
+        numrows = 0
+        still_querying = True
+        while still_querying:
+            request = self._build_request(current, max_rows)
+            resp = self._cb.post_object(url, body=request)
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._group_by_total_count = result["group_by_total_count"]
+
+            # Prevent 500 Internal Server Error from retrieving behind MAX_RESULTS_LIMIT
+            if self._total_results > MAX_RESULTS_LIMIT:
+                self._total_results = MAX_RESULTS_LIMIT
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                grouped_alert = self._doc_class(self._cb, None, item)
+                grouped_alert._request = self
+                yield grouped_alert
+                current += 1
+                numrows += 1
+
+                if max_rows > 0 and numrows == max_rows:
+                    still_querying = False
+                    break
+
+            from_row = current
+            if current >= self._total_results:
+                still_querying = False
+                break
+
+    def close(self, closure_reason=None, determination=None, note=None, ):
+        """
+        Closing all alerts matching a grouped alert query is not implemented.
+
+        Note:
+            - Closing all alerts in all groups returned by a ``GroupedAlertSearchQuery`` can be done by
+            getting the ``AlertSearchQuery`` and using close() on it as shown in the following example.
+
+        Example:
+            >>> alert_query = grouped_alert_query.get_alert_search_query()
+            >>> alert_query.close(closure_reason, note, determination)
```

Comment:
The order of the args is different, also do we need the trailing `,` is the parameters list?

Reply:
fixed both those.